### PR TITLE
Framework: Upgrade to eslint-plugin-wpcalypso v1.3.2

### PR DESCRIPTION
**Pull request by aduth**
This pull request seeks to upgrade our custom ESLint rules package from 1.1.3 to 1.3.2, including a bug fix and two new rules (`jsx-gridicon-size` and `jsx-classname-namespace`). The first of these replaces our existing `bin/check-gridicon-format` script and the second enforces our CSS namespace guidelines.

[See changelog for full details](https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/CHANGELOG.md)

__Testing instructions:__

The application should not be affected by this change. Note that running ESLint on problematic files details warnings for new and revised rules.

```
./node_modules/.bin/eslint client/layout/index.jsx 
```
```
/Users/andrew/Documents/Code/wp-calypso/client/layout/index.jsx
  196:23  warning  className should follow CSS namespace guidelines (expected layout__ prefix)  wpcalypso/jsx-classname-namespace
  201:24  warning  className should follow CSS namespace guidelines (expected layout__ prefix)  wpcalypso/jsx-classname-namespace
  202:26  warning  className should follow CSS namespace guidelines (expected layout__ prefix)  wpcalypso/jsx-classname-namespace

✖ 3 problems (0 errors, 3 warnings)
```

---

```
./node_modules/.bin/eslint client/components/author-selector/index.jsx 
```
```
/Users/andrew/Documents/Code/wp-calypso/client/components/author-selector/index.jsx
   83:64   warning  Gridicon size should be one of recommended sizes: 12, 18, 24, 36, 48, 54, 72  wpcalypso/jsx-gridicon-size
   92:61   warning  '?' should be placed at the beginning of the line                             operator-linebreak
   98:11   warning  ':' should be placed at the beginning of the line                             operator-linebreak
  100:119  warning  '?' should be placed at the beginning of the line                             operator-linebreak
  101:29   warning  ':' should be placed at the beginning of the line                             operator-linebreak
  106:8    warning  className should follow CSS namespace guidelines (expected author-selector)   wpcalypso/jsx-classname-namespace
  177:5    warning  JSX props should not use .bind()                                              react/jsx-no-bind

✖ 7 problems (0 errors, 7 warnings)
```

@mtias @johnHackworth @deBhal 

Test live: https://calypso.live/?branch=update/eslint-wpcalypso-1-3-x

### CODE REVIEW

File: npm-shrinkwrap.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
       "version": "1.0.3"
     },
     "abbrev": {
-      "version": "1.0.7"
+      "version": "1.0.9"
```
Comment (deBhal) [inline npm-shrinkwrap.json]: Why is shrinkwrap changing versions on modules we haven't touched?  I thought the point of it was to stop that happening?

Comment (aduth) [inline npm-shrinkwrap.json]: >Why is shrinkwrap changing versions on modules we haven't touched?

As I understand it, this is intentional, and _may_ be related to npm 3 dependency resolution ([docs](https://github.com/Automattic/wp-calypso/blob/master/docs/shrinkwrap.md#npm-3), cc @gwwar ).

Comment (gwwar) [inline npm-shrinkwrap.json]: So we lock down our top level dependencies, but sub-sub dependencies will update if for example nopt does not lock down their versions.
```
[kerryliu@gwwar ~/checkouts/wp-calypso] (master)$ npm ls abbrev
wp-calypso@0.17.0 /Users/kerryliu/checkouts/wp-calypso
├─┬ node-sass@3.4.2
│ └─┬ node-gyp@3.3.1
│   └─┬ nopt@3.0.6
│     └── abbrev@1.0.9 
└─┬ postcss-cli@2.5.1
  └─┬ chokidar@1.5.2
    └─┬ fsevents@1.0.12
      └─┬ node-pre-gyp@0.6.25
        └─┬ nopt@3.0.6
          └── abbrev@1.0.7 
```

Comment (gwwar) [inline npm-shrinkwrap.json]: npm 3 will also try to make the tree as flat as possible, and depends on install order for its structure: https://docs.npmjs.com/how-npm-works/npm3-nondet

**Comment by deBhal** [comment]
Thanks for the explanation.  It seems like that behaviour means that shrinkwrap doesn't fix the problem we're using it for completely, but clearly makes it much less bad.

The changes look good, but we needed to rebase anyway, so I've rebased, bumped the version up to 1.3.1 and re-run `make shrinkwrap`.

The conflicts were around make-shrinkwrap, so we may well need to run it again in a few minutes.

**Comment by aduth** [comment]
After resolving two edge cases, reflected in v1.3.1 and v1.3.2 of `eslint-plugin-wpcalypso` ([changelog](https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/CHANGELOG.md)), I think this is ready to go.

**Comment by deBhal** [comment]
:+1: Shows expected errors

**Pull request merged**